async-nats/src/jetstream/consumer.rs

```diff
@@ -0,0 +1,332 @@
+    /// subscriber. Pull-based consumers are intended to be used for
+    /// workloads where it is desirable to have a single process receive
+    /// a message. The only valid `ack_policy` for pull-based consumers
+    /// is the default of `AckPolicy::Explicit`, which acknowledges each
+    /// processed message individually. Pull-based consumers may be a
+    /// good choice for work queue-like workloads where you want messages
+    /// to be handled by a single consumer process. Note that it is
+    /// possible to deliver a message to multiple consumers if the
+    /// consumer crashes or is slow to acknowledge the delivered message.
+    /// This is a fundamental behavior present in all distributed systems
+    /// that attempt redelivery when a consumer fails to acknowledge a message.
+    /// This is known as "at least once" message processing. To achieve
+    /// "exactly once" semantics, it is necessary to implement idempotent
+    /// semantics in any system that is written to as a result of processing
+    /// a message.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub deliver_subject: Option<String>,
+
+    /// Setting `durable_name` to `Some(...)` will cause this consumer
+    /// to be "durable". This may be a good choice for workloads that
+    /// benefit from the `JetStream` server or cluster remembering the
+    /// progress of consumers for fault tolerance purposes. If a consumer
+    /// crashes, the `JetStream` server or cluster will remember which
+    /// messages the consumer acknowledged. When the consumer recovers,
+    /// this information will allow the consumer to resume processing
+    /// where it left off. If you're unsure, set this to `Some(...)`.
+    ///
+    /// Setting `durable_name` to `None` will cause this consumer to
+    /// be "ephemeral". This may be a good choice for workloads where
+    /// you don't need the `JetStream` server to remember the consumer's
+    /// progress in the case of a crash, such as certain "high churn"
+    /// workloads or workloads where a crashed instance is not required
+    /// to recover.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub durable_name: Option<String>,
+    /// A short description of the purpose of this consumer.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    /// Deliver group to use.
+    pub deliver_group: Option<String>,
+    /// Allows for a variety of options that determine how this consumer will receive messages
+    pub deliver_policy: DeliverPolicy,
+    /// Used in combination with `DeliverPolicy::ByStartSeq` to only select messages arriving
+    /// after this sequence number.
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        rename = "opt_start_seq"
+    )]
+    pub opt_start_sequence: Option<u64>,
```

Comment:
We should roll this into the `DeliverPolicy` enum, I need a refresher on how we can do that with serde but it is possible.

```suggestion
    pub opt_start_sequence: Option<u64>,
```

Can follow up with it.